### ISSUE TITLE
Throw an error on unrecognized pl-* elements

### ISSUE THREAD

**Issue by mwest1066** [PrairieLearn/PrairieLearn#2279]
In `traverseQuestionAndExecuteFunctions()` inside `freeform.js`, if `visitNode()` finds an element with `tagName` starting with `pl-` that we don't recognize, then it should throw an error rather than silently proceeding.

**Comment by eliotwrobson** [comment]
I think that implementing this is actually somewhat problematic. The traversal function traverses the inner `pl` tags as well as the outer ones, but the platform doesn't have information about which inner tags should be recognized. It could retrieve these from a hardcoded list somewhere, but this seems like overcomplicating things for no _huge_ benefit.

**Comment by echuber2** [comment]
I agree this may be a problem if course staff had intentionally been using `pl-` for course elements (or something similar). Although the server could conceivably include course elements in the list of "known" tag names, perhaps course staff are doing something unusual. Would it make sense to make this optional with a "strict" setting on the course? Or have a way to explicitly add tag names to ignore to the course?

**Comment by echuber2** [comment]
On second thought, I haven't looked at the renderer in detail recently. If the expectation is that every `pl-` tag should be recognized and converted at that stage (so the other action would be to discard it silently), then it seems better to error than to discard it. If the current action is to leave it alone (to be converted in another pass), then erroring out seems overzealous.

**Comment by nwalters512** [comment]
IIRC the traversal algorithm always processes parents before children, so even if you have a `pl-answer` inside a `pl-checkbox`, the renderer itself should never see that child element as it will have been turned into something else when `pl-checkbox` renders. This should mean that we don't have to know about any "inner" elements like that.

**Comment by echuber2** [comment]
Maybe I'm not reading your comment correctly, but did you mean it processes children before parents?

**Comment by nwalters512** [comment]
Nope, I meant parents first. Studying the implementation is definitely recommended!

**Comment by eliotwrobson** [comment]
@nwalters512 is this the type of warning that would be better suited to being behind a feature flag? Looking at this again, I'm wondering if bundling this with the Python renderer is the way to go (as I did in #7983), since the new renderer is still opt-in only. Having a feature flag for just this warning feels like a very small change to put behind something like that, and enabling it for all courses at once is probably going to throw a ton of errors.
